package.json

```diff
@@ -30,6 +31,7 @@
     "node": ">=4"
   },
   "dependencies": {
+    "@types/jquery": "^2.0.39",
```

Comment:
Yes, I think it would do. In this case, optionalDependencies to my knowledge will install the dependency anyway, I won't install only if the @type dependency itself fails. But sure it will work.
It's also okay to put it in devDependencies.